### ISSUE TITLE
RoomTagGenerator Class

### ISSUE THREAD

**Issue by mattpadgett** [Tervoc/roomee-api#2]
Utility RoomTagGenerator class issues:

- Class variable "random" has a non-descriptive name and does not follow the C# variable naming standard.
- Why does the "FindUnusedTag" function take in a length? Is this necessary for your application or will they all be a standard length? This is a discussion item. Not saying it's wrong or right.
- "TagIsTaken" has a slightly unclear name. Generally speaking for functions like this it's good to put the "Is" first. So, "IsTagTaken". Basically, just phrased as a question rather than a statement because you are returning a boolean value.
- "TagIsTaken" logic only checks if a tag has ever been used. Would you possibly want the ability to reuse a tag after they expire or are you going to have it to where users can reactivate expired tags?

**Comment by Tervoc** [comment]
- that variable and others that did not follow the standard closely enough have been changed and will be included in the pull request for the room-auth branch 
- Valid point. it was a part of the reference code I used for the function and left it in as a part of the base funcitonality of the function. Since we will be using the same length tags for all room tags a standard tag length and removal of parameter intake will be a part of the pull request for room-auth 
- the idea was that in the FindUnusedTag function the line where the aforementioned function is used would read as "if the tag is taken" but sticking to a standard makes more sense than a one-line gimmick 
- reusing tags creates conflict between those who created the original tag and those who re use the tag. a user might have a deprecated or expired tag to a room they wish join but instead find themselves joining a completely different room. Re-activating expired tags is a feature that is to be implemented. Since we will not be re using tags for different rooms there is no ambiguity to which room the expired tag will be re assigned to and to which room a user will join using said tag.
